### PR TITLE
Support PvP rank filtering in IV filter

### LINKED ISSUE #212: PVP-Or does not work
As expected, since it's never been implemented.

### PR DESCRIPTION
**Pull request by Mygod**
Fixes #212. Also fixes the issue where @versx cannot write a correct SQL when there are more than 8 clauses in a WHERE.

Usage: `GR1-3` for great league rank 1-3.

Needs testing.

For future:

1. Currently we use the best evolution rank as the rank for the Pokemon. This is not ideal for e.g. Eevee. A possible solution is to add a `PREEV` keyword to IV filter to toggle whether we are looking at a preevolution.
2. We should display the rank only if the Pokemon at level 41 maxes below 1500/2500.

Both of these involves changes in DataParser.

### CODE REVIEW

File: src/data/map.js [hunk 1]
```diff
@@ -1258,12 +1190,14 @@ const sqlifyIvFilter = (filter) => {
                     case 'S': column = 'sta_iv'; break;
                     case 'L': column = 'level';  break;
                     case 'CP': column = 'cp';    break;
+                    case 'GR': column = 'great_rank'; break;
+                    case 'UR': column = 'ultra_rank'; break;
```
Comment (versx) [inline src/data/map.js:1194]: If you don't have plans to add support for filtering percentage or any other properties I think `GL` and `UL` would be better keywords personally.

Comment (sketchysocks) [reply src/data/map.js:1194]: Agreed

Comment (Mygod) [reply src/data/map.js:1198]: DiSaGrEeD

**Comment by sketchysocks** [comment]
- This one seems to be functioning as anticipated now that the bugs have been addressed, just a couple of comments:
- Before merging I think we should discuss the possibility of renaming IV-AND/IV-OR to more general terms since IVs are now only a small part of the filter. 
- GR and UR should be changed to GL and UL. I don't believe it's necessary to add % down the road so GL/UL are fine and the most familiar acronyms for users.

**Comment by Mygod** [comment]
You can do those in a separate PR I guess.

Could someone send me some profiling data? This thing shouldn't be this slow.

**Comment by clburlison** [comment]
The changes are working great over here. I'm not noticing a noticeable slow down between this branch and master. Tried doing an
IV-Or = 100  plus   PVP-AND = 1-10

Then
ur1-10 | gr1-10 | 100

As far as my browser was concerned these had the same load times.

**Comment by Mygod** [comment]
I'm seeing a ~6x slowdown. It should only be noticeable when you zoom out.

File: src/data/map.js (hunk 1 above)
Comment (clburlison) [reply src/data/map.js:1202]: Basically all mapping tools reference this as Great/Ultra League pokemon. The rank is just the text value. Changing to GL/UL just matches better. Agreed with versx & sketchysocks.

Users aren't filtering Great Rank Pokemon 1-3. They're filtering Great League Pokemon 1-3.

Comment (Mygod) [reply src/data/map.js:1202]: Ok fine.

**Review by clburlison — APPROVED** [review]
I'm not seeing the performance penalty. Good to merge from my testing.

**Comment by Mygod** [comment]
Okay I guess in the realistic setting (not zooming all the way out) the slowdown is not noticeable at all.

**Review by sketchysocks — APPROVED** [review]

**Pull request merged**